tests/optimizers/tests_optimizers.py

```diff
@@ -101,10 +101,10 @@ def test_adagrad():
         #trainer.append_task(tasks.PrintVariable("Loss gradient: {}", loss.gradients[param]))
 
         # Monitor the gradient of `loss` w.r.t. to `param`.
-        gparam = tasks.MonitorVariable(loss.gradients[param])
-        trainer.append_task(gparam)
+        gparam = views.MonitorVariable(loss.gradients[param])
+        trainer._graph_updates.update(gparam.updates)
```

Comment:
Another thing we could do is pass a variable directly, then the `Tracker` would create the view itself.

```python
def __init__(self, *vars, **freqs):
  super().__init__(*[views.MonitorVariable(v) for v in vars], **freqs)
```